Tasks/TerraformTask/TerraformTaskV1/src/base-terraform-command-handler.ts

```diff
@@ -119,16 +119,23 @@ export abstract class BaseTerraformCommandHandler {
             "plan",
             tasks.getInput("workingDirectory"),
             tasks.getInput(serviceName, true),
-            tasks.getInput("commandOptions")
+            `${tasks.getInput("commandOptions")} -detailed-exitcode`
         );
         
         let terraformTool;
         terraformTool = this.terraformToolHandler.createToolRunner(planCommand);
         this.handleProvider(planCommand);
     
-        return terraformTool.exec(<IExecOptions> {
-            cwd: planCommand.workingDirectory
+        let result = terraformTool.exec(<IExecOptions> {
```

Comment:
need to await result  
```suggestion
        let result = await terraformTool.exec(<IExecOptions> {
```

Reply:
Updated PR based on comments